lib/rspec/core/hooks.rb

```diff
@@ -455,7 +455,7 @@ def register(prepend_or_append, position, *args, &block)
         def run(position, scope, example_or_group)
           return if RSpec.configuration.dry_run?
 
-          if scope == :context
+          if scope == :context && !example_or_group.class.metadata[:skip]
```

Comment:
I believe there is a bug here.  Consider what happened before your change when `scope == :context`: it _never_ executed the `else` clause code.  Consider what happens now when the group is not tagged with `:skip`: the 2nd part of the conditional evaluates to false, and the `else` clause code will get executed.  That will potentially result in `before(:example)` hooks running an extra time (when RSpec is intending to run `before(:context)` hooks).  So I think this needs to be:

``` ruby
if scope == :context
  unless example_or_group.class.metadata[:skip]
    # ...
  end
else
  # ...
end
```